### PR TITLE
DON-384: Adjust the TextField of the Place Selector GC 

### PR DESCRIPTION
**Pull request by yurareutskiy**
Last minute changes from a design team:
- change default icon
- update placeholder colour
- hide a close button if it's empty

+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] `README.md`
+ [ ] Tests
+ [ ] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [ ] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack-SwiftUI/SearchInputSummary/Classes/BPKSearchInputSummary.swift [hunk 1]
```diff
@@ -93,6 +94,12 @@ public struct BPKSearchInputSummary: View {
         }
     }
     
+    @ViewBuilder
+    private var placeholderView: Text {
+        Text(placeholder)
+            .foregroundColor(Color(BPKColor.textSecondaryColor.value))
```
Comment (frugoman) [inline Backpack-SwiftUI/SearchInputSummary/Classes/BPKSearchInputSummary.swift:100]: Should you not be using a BPKText here?
```swift
BPKText(placeholder).foregroundColor(.textSecondary)

File: Backpack-SwiftUI/SearchInputSummary/Classes/BPKSearchInputSummary.swift [hunk 2]
```diff
@@ -58,14 +58,15 @@ public struct BPKSearchInputSummary: View {
...
-            TextField(placeholder, text: $text)
+            TextField(placeholder, text: $text, prompt: placeholderView)
                 .font(style: .bodyDefault)
                 .foregroundColor(state.textColor)
                 .disabled(state.isDisabled)
                 .focused($focused)
                 .accessibilityElement(children: .combine)
                 .accessibilityAddTraits(.isSearchField)
             accessory
+                .opacity(text.isEmpty ? 0.0 : 1.0)
```
[the remark this answers is not in the document]
Comment (frugoman) [reply Backpack-SwiftUI/SearchInputSummary/Classes/BPKSearchInputSummary.swift:69]: Also, this assumes that the acecssory is the clear button, but the accessory icon can be something else other than a clear button, i think this should be moved below, where you create the accessory clear button only.
as in.. you may not want the accessory to disaperar if it's not a clear buton

[the remark this answers is not in the document]
Comment (yurareutskiy) [reply Backpack-SwiftUI/SearchInputSummary/Classes/BPKSearchInputSummary.swift:69]: Accessibility works good, it seems SwiftUI doesn't render the element if it has 0 opacity. I've checked it :) 
Good point about moving it to close button specific.

File: Backpack-SwiftUI/SearchInputSummary/Classes/BPKSearchInputSummary.swift (hunk 1 above)
Comment (yurareutskiy) [reply Backpack-SwiftUI/SearchInputSummary/Classes/BPKSearchInputSummary.swift:100]: No, it's accepted only Text view. It won't compile with BPKText

**Review by frugoman — APPROVED** [review]

**Pull request merged**